pkg/flow/internal/controller/loader.go

```diff
@@ -616,6 +616,8 @@ func (l *Loader) EvaluateDependencies(c *ComponentNode) {
 		}()
 
 		var err error
+		// Record the time the dependencies had to wait to be evaluated.
+		l.cm.dependenciesWaitTime.Observe(time.Since(c.lastUpdateTime.Load()).Seconds())
```

Comment:
Yep, it's a starting component that triggers dependencies sub-graph walk-through. To be more precise, I could change this to be time since its parent is evaluated - would require some small changes to the walk function, but is doable.